symphony/bdk/core/extension.py

```diff
@@ -0,0 +1,54 @@
+import logging
+from abc import ABC, abstractmethod
+
+from symphony.bdk.core.auth.auth_session import AuthSession
+from symphony.bdk.core.config.model.bdk_config import BdkConfig
+
+logger = logging.getLogger(__name__)
+
+
+class BdkConfigAware(ABC):
```

Comment:
Event if we use duck typing, I think it is nice to have abstract classes to document in the code which interface we expect.

Reply:
⚠️ For the retry mechanism, we use `@retry` decorators on methods that needs to be retried. This decorator checks retry configuration in field `self._retry_config` assuming such field exists. 
We will have to properly document that.